File: longestIncreasingSubsequence.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        if not nums:
            return 0
        memo = [1] * len(nums)
        for i in range(1, len(nums)):
            for j in range(i):
                if memo[i] > memo[j]:
                    continue
                if nums[i] > nums[j]:
                    memo[i] = memo[j] + 1
        return max(memo)

File: longestIncreasingSubsequence.js
function longestIncreasingSubsequence(seq){
	let memo = new Array(seq.length);
	// The least longest increasing subsequence is 1 for each element
	memo.fill(1);
	for(let i = 1; i < seq.length; i++){
		for(let j = 0; j < i; j ++){
			if(memo[i] > memo[j]) continue;
			if(seq[j] < seq[i]) memo[i] = memo[j] + 1;
		}
	}
	return Math.max.apply(null, memo)
}
